### PR TITLE
Introduce SetEditor, Nestable Editors

### LINKED ISSUE #3561: Need a mechanism to defer edits to graphs of noms data and commit when edits are complete.

### PR DESCRIPTION
**Pull request by rafael-atticlabs**
Tests for nested editors coming. Please review now.

@aboodman 

cc @kalman 

Fixes #3561

### CODE REVIEW

**Comment by aboodman** [comment]
Title of PR is wrong.

File: go/types/set_editor.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+func (se *SetEditor) Kind() NomsKind {
+	return SetKind
+}
+
+func (se *SetEditor) build(vrw ValueReadWriter) Value {
+	return se.Build(vrw)
+}
+
+func (se *SetEditor) Build(vrw ValueReadWriter) Set {
```
Comment (aboodman) [inline go/types/set_editor.go]: What do you think about calling this `Set()` (and `Map()` in the case of `MapEditor`). It is more parallel with the `Value()` method which does the same thing.

File: go/types/map_editor.go [hunk 2]
```diff
@@ -38,14 +47,56 @@ func (me *MapEditor) Build(vrw ValueReadWriter) Map {
...
+				continue // next edit supercedes this one
+			}
+
+			// TODO: Use ReadMany
+			cc := make(chan *sequenceCursor, 1)
+			cursChan <- cc
+
+			go func(k Value, cc chan *sequenceCursor) {
+				cc <- newCursorAtValue(me.m.seq, k, true, false, false)
+			}(edit.key, cc)
```
Comment (aboodman) [inline go/types/map_editor.go]: There is no need to pass `cc` -> it's already local to each time around the loop. And wrt `edit.key`, is more idiomatic in Go to reassign to a local:

```
for .... {
  edit := edit
  ...
}
```

Comment (rafael-atticlabs) [inline go/types/map_editor.go]: done

File: go/types/set_editor.go (hunk 1 above)
Comment (rafael-atticlabs) [inline go/types/set_editor.go]: ok

**Pull request merged**